http/server.ts

```diff
@@ -215,10 +215,10 @@ export class ServerRequest {
   }
 }
 
-async function readRequest(
+export async function readRequest(
   conn: Conn,
   bufr: BufReader
-): Promise<[ServerRequest, BufState]> {
+): Promise<[ServerRequest, BufState | Error]> {
```

Comment:
I think the `BufState` already allows for `Error` ?

Reply:
You're right. Fixing this.